File: fs/Practice2/Mathematical/GreedyTakahashi.fs
// https://atcoder.jp/contests/abc149/tasks/abc149_b

stdin.ReadLine().Split() |> Array.map int64 |> fun x -> (x.[0], x.[1], x.[2])
|> fun (a, b, k) ->
  (max 0L (a - k), max 0L (b - (max 0L (k - a))))
|> fun (a, b) ->
  printfn "%d %d" a b

// https://atcoder.jp/contests/abc149/submissions/29734226

File: pl/Practice2/Mathematical/GreedyTakahashi.pl
% https://atcoder.jp/contests/abc149/tasks/abc149_b

read_string(Str) :-
  current_input(Input),
  read_string(Input, ' \n', '', _, Str).

read_number(Num) :-
  read_string(Str),
  number_string(Num, Str).

solve(A, B, K, AR, BR) :-
  AR is max(0, A - K),
  M  is max(0, K - A),
  BR is max(0, B - M).

main :-
  read_number(A),
  read_number(B),
  read_number(K),
  solve(A, B, K, AR, BR),
  concat_atom([AR, BR], ' ', Ret),
  write(Ret).

:- main.

% https://atcoder.jp/contests/abc149/submissions/29920631
